apps/condo/domains/resident/utils/serverSchema/findOrganizationsByAddress.js

```diff
@@ -11,112 +10,206 @@ const {
+    const billingAccounts = await find('BillingAccount', {
+        context: { id: billingContext.id },
+        property: { addressKey, deletedAt: null },
+        deletedAt: null,
+        ...query,
+    })
+    let receipts = []
+
+    if (billingAccounts.length && get(billingContext, 'lastReport.period')) {
+        receipts = await find('BillingReceipt', {
+            context: { id: billingContext.id },
+            property: { addressKey, deletedAt: null },
+            deletedAt: null,
+            period: billingContext.lastReport.period,
+            account: { id_in: billingAccounts.map(({ id }) => id) },
         })
-        return meter ? id : null
-    }))
-    return new Set(organizationsWithMeters.filter(Boolean))
+
+        const billingAccountsNumbersIndex = Object.fromEntries(
+            billingAccounts.map(account => [account.id, account.number])
+        )
+        receipts = receipts.map(receipt => ({
+            category: receipt.category,
+            balance: receipt.toPay,
+            accountNumber: billingAccountsNumbersIndex[receipt.account],
+            routingNumber: receipt.recipient.bic,
+            bankAccount: receipt.recipient.bankAccount,
+            address: get(receipt, 'raw.address', null),
+        }))
+    }
+
+    return receipts
+}
+
+async function getOrganizationMeters (organization, addressKey, query = {}, properties) {
+    let meters = await find('Meter', {
+        organization: { id: organization.id },
+        deletedAt: null,
+        property: { addressKey, deletedAt: null },
+        ...query,
+    })
+
+    const property = properties.find(p => p.addressKey === addressKey)
+    const address = property ? property.address : null
+
+    if (meters.length){
+        const meterReadings = await find('MeterReading', {
+            meter: { id_in: meters.map(({ id }) => id) },
+            deletedAt: null,
+        })
+        const meterReadingIndex = meterReadings.reduce((acc, { meter, value1, value2, value3, value4 }) => {
+            acc[meter] = [value1, value2, value3, value4].filter(Boolean).join(',')
+            return acc
+        }, {})
+        meters = meters.map((meter) => ({
+            resource: meter.resource,
+            accountNumber: meter.accountNumber,
+            number: meter.number,
+            value: meterReadingIndex[meter.id] || null,
+            address,
+        }))
+    } else {
+        meters = null
+    }
+
+    return meters
 }
 
-async function getOrganizationsWithAcquiring (organizationIds) {
+async function getOrganizationBillingContext (organization) {
+    const [context] = await find('BillingIntegrationOrganizationContext', {
+        status: BILLING_CONTEXT_FINISHED_STATUS,
+        deletedAt: null,
+        organization: { id: organization.id },
+    })
+
+    return context
+}
+
+async function getOrganizationIdsWithMeters (organizations) {
+    const meterResourceOwners = await find('MeterResourceOwner', {
```

Comment:
That's a lot of data... ~`properties.length * organizations.length`

Reply:
Why do even need this util? It has no relation to addressKey at all....Unused logic